Gems/EMotionFX/Code/Source/Integration/Components/ApplyMotionSetComponent.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#if defined(CARBONATED)
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TransformBus.h>
+#include <AzCore/Component/TickBus.h>
+#include <AzCore/Asset/AssetCommon.h>
+#include <AzCore/Script/ScriptProperty.h>
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+
+#include <Integration/Assets/MotionSetAsset.h>
+#include <Integration/ApplyMotionSetComponentBus.h>
+#include <AzCore/Component/EntityId.h>
+
+namespace EMotionFX
+{
+    namespace Integration
+    {
+        class ApplyMotionSetComponent
+            : public AZ::Component
+            , private AZ::Data::AssetBus::MultiHandler
+            , private ApplyMotionSetComponentRequestBus::Handler
+        {
+        public:
+
+            friend class EditorApplyMotionSetComponent;
+
+            AZ_COMPONENT(ApplyMotionSetComponent, "{1B4ED2C1-58F5-44A4-BF6E-C22667AC60CD}");
+
+            /**
+            * Configuration struct for procedural configuration of Actor Components.
+            */
+            struct Configuration
+            {
+                AZ_TYPE_INFO(Configuration, "{C1DD0FAF-0DEA-4965-940A-0E8A3FE8EABD}")
+
+                Configuration();
+
+#if defined(CARBONATED)

```

Comment:
Looks like inner #ifdef/#endif for CARBONATED are not necessary
here and below
due to we use #if defined(CARBONATED) for whole file in the line 3
?